### PR TITLE
LZ4 compression for import/export

### PR DESCRIPTION
**Pull request by bharatnc**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Introduce lz4 compression for import / export.

Detailed description / Documentation draft:

This PR adds lz4 compression for import/export operations. 

Related to: #25171

### CODE REVIEW

File: cmake/find/lz4.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+if(NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/lz4/lib/lz4.h")
+    if(USE_LZ4)
+        message(WARNING "submodule contrib/lz4 is missing. to fix try run: \n git submodule update --init --recursive")
+        message (${RECONFIGURE_MESSAGE_LEVEL} "Can't find internal lz4 library")
+        set(USE_LZ4 0)
+    endif()
+    set(MISSING_INTERNAL_LZ4_LIBRARY 1)
+endif()
+
+if (NOT USE_LZ4)
```
Comment (filimonov) [inline cmake/find/lz4.cmake:12]: I don't think clickhouse is ready to work / build w/o LZ4.

Comment (bharatnc) [reply cmake/find/lz4.cmake:12]: Ok - In that case, I guess that I'll always build it with Lz4.

**Comment by evillique** [comment]
Are you planning on finishing this pull request? I can help you finish it, if you don't mind :)

**Comment by bharatnc** [comment]
> Are you planning on finishing this pull request? I can help you finish it, if you don't mind :)

@evillique sure that would be helpful. I haven't been able to get to this and I wasn't able to get the lz4 api decompression to work with Clickhouse - was facing some issues. It might take a bit for me to work on this again and don't want to block the development of this feature. You can pick this up. Thank you!

File: tests/queries/0_stateless/01059_storage_file_brotli.sql [hunk 2]
```diff
@@ -31,10 +31,19 @@ SELECT count(), max(x) FROM file;
 
 DROP TABLE file;
 
-SELECT count(), max(x) FROM file('data{1,2,3,4}.tsv.{gz,br,xz,zst}', TSV, 'x UInt64');
+CREATE TABLE file (x UInt64) ENGINE = File(TSV, 'data5.tsv.lz4');
+TRUNCATE TABLE file;
+
+INSERT INTO file SELECT * FROM numbers(1000000);
```
Comment (bharatnc) [inline tests/queries/0_stateless/01059_storage_file_brotli.sql:37]: This comment was supposed to belong to this PR but I guess I ended up commenting elsewhere:

@evillique I've noticed this behavior (before as well while I was doing the initial implementation) where inserting into file / table with large number of rows causes the query to hang. So I added this test (similar to other compression methods) to confirm and it still hangs according to my local test.

Can also reproduce using (query never completes):

tower :) insert into function file('test2.lz4', 'TSV', 'number UInt64', 'lz4') select * from numbers(0,1000000);

INSERT INTO FUNCTION file('test2.lz4', 'TSV', 'number UInt64', 'lz4') SELECT *
FROM numbers(0, 1000000)

Query id: 3fa90fbe-9bab-4d92-9aaf-79763ddc5de1

→ Progress: 1.00 million rows, 8.00 MB (9.12 million rows/s., 72.99 MB/s.)  99%

But after your changes in f2852d1a088ee55fcbf6bd9c17dfb742edf0ff8a, it now works fine.

**Review by evillique — APPROVED** [review]

**Comment by bharatnc** [comment]
Thank you @evillique !

**Comment by sevirov** [comment]
Internal documentation ticket: DOCSUP-19959

**Pull request merged**